lib/closex/mock_client/mock_client.ex

```diff
@@ -0,0 +1,246 @@
+defmodule Closex.MockClient do
+  @behaviour Closex.ClientBehaviour
+
+  @moduledoc """
+  This module is for testing, it allows you to stub requests to CloseIO.
+
+  It behaves like the standard client so you can drop it into your code via
+  configuration in your testing environment.
+
+  In test mode you can set the MockClient:
+
+  ```
+  $ cat your_app/config/test.exs
+
+  config :yourapp,
+    closeio_client: Closex.MockClient,
+    ...other configuration...
+  ```
+
+  The mock client will return the test values for each method.
+
+  For more detail on the test objects see the `test/fixtures/*.json` files.
+
+  However, you can override the response of the mock client with json more specific to your domain:
+
+  ```
+  $ cat your_app/config/test.exs
+
+  config :yourapp,
+    closeio_client: Closex.MockClient,
+    ...other configuration...
+
+  config :closex,
+    mock_client_fixtures_dir: Path.join([File.cwd!, "path", "to", "your", "fixtures"])
+  ```
+
+  If you'd like to emulate not finding an object, pass in the `Closex.MockClient.get_not_found_id` value.
+  """
+
+  @not_found_id "not_found"
+  def get_not_found_id do
+    @not_found_id
+  end
+
+  @lead_id "lead_IIDHIStmFcFQZZP0BRe99V1MCoXWz2PGCm6EDmR9v2O"
+  def get_lead_id do
```

Comment:
Rename to `lead_id` etc

Reply:
👍  Was still on my list of stuff to do once I had finished all the changes, done now.